src/v/bytes/bytes.h

```diff
@@ -46,6 +47,15 @@ struct bytes_type_hash {
     size_t operator()(const bytes_view&) const;
 };
 
+
+struct bytes_type_murmur_hash {
```

Comment:
I think this should be a template. 

```
template<typename Hasher>
struct bytes_hasher {
  std::size_t operator() (const bytes_view& b) const {
        return Hasher(b.data(), b.size());
   }
   size_t operator() (const bytes& bb) const { return operator()(bytes_view(bb)); }
};
```

Reply:
then we can use murmur, xxhash, jenkins, fnv1a, etc

Reply:
`Hasher` must be a type, since we do not have `Hasher` structs for each hash type, we would have to create one for each hash f.e.

```c++
struct xx64_bytes_hasher {
    size_t operator()(bytes::const_pointer data, size_t len) {
        return XXH64(data, len, 0);
    }
};
```
@senior7515 
Do you think this is ok ?

Reply:
just create a template type then. 

```
template<typename Func>
struct foo {
     using fn = Func;
    size_t operator()(const bytes_view& b) {
          return fn(b.data(), b.size());
    }
};```

or smth like that no? i think it can be generalized.